### PR TITLE
Generator: Fix additional properties spec 3.1

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description

## Checklist
(For exoscale contributors)

* [ ] Changelog updated (under *Unreleased* block)
* [ ] For a new resource or new attributes: test added/updated

### CODE REVIEW

**Review by kobajagi — APPROVED** [review]

**Comment by pierre-emmanuelJ** [comment]
It why I did this dirty humanize integer workaround:
image
to prevent name collision....
And also if the value is not alphanumeric, I print the index of the enum instead of the value in the const name suffix

that's two workaround to make the spec compile

File: v3/generator/schemas/schemas.go [hunk 1]
```diff
@@ -526,3 +537,22 @@ func isNullableReference(node *yaml.Node) bool {
...
+	if err != nil {
+		return s
+	}
+
+	if n >= 0 {
+		return s
+	}
+
+	// remove the "-" with helper
+	return helpers.ToLowerCamel(fmt.Sprintf("minus%s", s))
```
Comment (sauterp) [inline v3/generator/schemas/schemas.go:557]: why is ToLowerCamel still needed here?

Comment (pierre-emmanuelJ) [reply v3/generator/schemas/schemas.go:557]: it's dropping the char "-" to make it lower camel: "minus-1" become "minus1"

**Review by kobajagi — APPROVED** [review]

File: v3/generator/schemas/schemas.go [hunk 2]
```diff
@@ -203,16 +204,32 @@ func renderSchemaInternal(schemaName string, s *base.Schema, output *bytes.Buffe
 }
 
 func renderSimpleTypeEnum(typeName string, s *base.Schema) string {
+	for _, e := range s.Enum {
+		if s.Minimum != nil || s.Maximum != nil {
+			return ""
+		}
+
```
Comment (sauterp) [inline v3/generator/schemas/schemas.go:211]: When the new public-api with the sanitized spec is published, you may want to delete these lines again and regenerate before merging.

**Review by sauterp — APPROVED** [review]

**Pull request merged**